pandas/core/frame.py

```diff
@@ -645,6 +645,23 @@ def _constructor(self) -> Callable[..., DataFrame]:
 
     _constructor_sliced: Callable[..., Series] = Series
 
+    def __init_subclass__(cls, /, **kwargs) -> None:
+        super().__init_subclass__(**kwargs)
+        if cls._constructor is not DataFrame._constructor:
+            warnings.warn(
+                "pandas subclass support is deprecating allowing _constructor "
+                "that does not return a class.",
+                FutureWarning,
+                stacklevel=find_stack_level(),
+            )
```

Comment:
does this raise a warning all the time, with no option to silence it?

just wondering if there's a way to do this without having to always raise a warning, and if not, if the warning should advise to just use `filterwarnings` to suppress the message because there's no workaround

Reply:
> does this raise a warning all the time, with no option to silence it?

Yes.  I tried having it check and only raise if _constructor_foo doesn't return a class but couldn't get that working.

> just wondering if there's a way to do this without having to always raise a warning, and if not, if the warning should advise to just use filterwarnings to suppress the message because there's no workaround

Makes sense.  Or as mentioned in the OP this could be doc-only.